azavea/franklin
# Update API to 0.9.0

## Overview

This upgrades the API to STAC version 0.9.0 and upgrades dependencies.

### Checklist

~- [ ] New tests have been added or existing tests have been modified~

### Notes

I made a few decisions here that are worth highlighting.

First, in the issue for the upgrade there is mention about removing the `next` parameter from the `Context` object (formerly `search:metadata`). However, I chose not to do that since we have not yet implemented token based pagination. I updated that issue to note that in the process of implementing pagination that should be removed. It's not a big change to make, but if we do it without implementing pagination as specified in the API spec then there will be no way to support it. So for now, I left it.

Second, I created an example catalog of Sentinel 2 data because I needed a catalog to test with.

### Testing Instructions

- Look over #41 and make sure I made the changes I needed to make
- Clear database
```
docker-compose up -d database
./scripts/dbshell
truncate table collection_items; truncate table collections;
```
- Load Sentinel 2 Sample Catalog:
```
bloop run application -- import --catalog-root s3://rasterfoundry-development-data-us-east-1/catalog.json  --db-user franklin --db-password franklin --db-host localhost --db-name franklin
```
- Start Server: `./scripts/server`
- Test some endpoints:
 - http://localhost:9090/search <- notice the root is gone, this is search results -- datetime filtering, etc. should work here
 - http://localhost:9090/ <- notice I fixed the spec link in here

Closes #41
